### PR TITLE
soc: arm: st_stm32: add STM32F437XI

### PR DESCRIPTION
**Pull request by mrfuchs**
This patch adds support for the STM32F437XI SoC.

Signed-off-by: Markus Fuchs <markus.fuchs@de.sauter-bc.com>

### CODE REVIEW

File: dts/arm/st/f4/stm32f437.dtsi [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2019, Markus Fuchs <markus.fuchs@de.sauter-bc.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/f4/stm32f407.dtsi>
```
Comment (erwango) [inline dts/arm/st/f4/stm32f437.dtsi]: STM32F437 indeed inherits from STM32F407 but supports as well (among the IPs currently supported in zephyr):
-UART7 and 8
-SPI4, 5 and SPI6
-GPIOJ and K
Please add them here.

File: soc/arm/st_stm32/stm32f4/Kconfig.defconfig.stm32f437xx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+config GPIO_STM32_PORTF
+	default y
+
+config GPIO_STM32_PORTG
+	default y
+
+config GPIO_STM32_PORTI
+	default y
+
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f4/Kconfig.defconfig.stm32f437xx]: Add ports J and K here as well

**Review by erwango — CHANGES_REQUESTED** [review]
Few comments, otherwise sounds good

File: dts/arm/st/f4/stm32f437.dtsi (hunk 1 above)
Comment (mrfuchs) [reply dts/arm/st/f4/stm32f437.dtsi]: Thanks. I have added these. I used stm32f429.dtsi as a template. As a side note, is it correct that stm32f429.dtsi defines i2s4, i2s5 and i2s6? From the documentation, it looks like there is only i2s2 and i2s3.

File: soc/arm/st_stm32/stm32f4/Kconfig.defconfig.stm32f437xx (hunk 2 above)
Comment (mrfuchs) [reply soc/arm/st_stm32/stm32f4/Kconfig.defconfig.stm32f437xx]: Added - thanks.

File: dts/arm/st/f4/stm32f437.dtsi (hunk 1 above)
Comment (erwango) [reply dts/arm/st/f4/stm32f437.dtsi]: IIRC, i2s4 5 and 6 are supported on all stm32f429 variants except the stm32f429**v**X ones.
So we delete them in stm32f429vX.dtsi file.

Comment (erwango) [reply dts/arm/st/f4/stm32f437.dtsi]: Looking more into it, seems you're right about I2S instances. I'll fix that .

**Review by erwango — APPROVED** [review]

**Comment by ioannisg** [comment]
@mrfuchs @erwango just to clarify: this patch adds SOC support but there is no board yet, right?

**Comment by mrfuchs** [comment]
> @mrfuchs @erwango just to clarify: this patch adds SOC support but there is no board yet, right?

That's right. I created these files for a custom board.

**Review by ioannisg — APPROVED** [review]

**Pull request merged**